src/System.IO.FileSystem/src/System/IO/PathHelpers.Windows.cs

```diff
@@ -87,9 +82,7 @@ internal static void SplitDirectoryFile(string path, out string directory, out s
         internal static string NormalizeSearchPattern(string searchPattern)
         {
             Debug.Assert(searchPattern != null);
-
-            // Win32 normalization trims only U+0020.
-            string tempSearchPattern = searchPattern.TrimEnd(PathHelpers.TrimEndChars);
+            string tempSearchPattern = searchPattern;
```

Comment:
May as well reduce this to:

`string tempSearchPattern = searchPattern.Equals(".") ? "*" : searchPattern;`